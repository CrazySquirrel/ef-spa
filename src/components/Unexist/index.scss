@import "../../scss/base";

.unexist {
  width: 100%;
  height: 100%;
  background: $color6 url('../../images/cat.gif') center center no-repeat;

  &__title {
    color: $color7;
    position: absolute;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
    margin: 100px 0 0 0;
  }
}