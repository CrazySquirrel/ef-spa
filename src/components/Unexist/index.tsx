import * as React from 'react';
import * as bem from 'bem-cn';

import {Metadata} from '../../types';

interface Props extends Metadata {
}

import './index.scss';

export default class Unexist extends React.Component<Props, {}> {
  public render() {
    const block = bem('unexist');

    return (
        <section className={block()}>
          <h1 className={block('title')()}>{this.props.h1}</h1>
        </section>
    );
  }
}
