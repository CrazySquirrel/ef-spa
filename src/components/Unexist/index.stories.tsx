import * as React from 'react';

import {storiesOf} from '@storybook/react';
import {withKnobs, text} from '@storybook/addon-knobs';

import Unexist from './index';

const stories = storiesOf('Unexist', module);

stories.addDecorator(withKnobs);

stories.add('default', () => (
    <Unexist h1={text('H1', 'H1 test')}/>
));
