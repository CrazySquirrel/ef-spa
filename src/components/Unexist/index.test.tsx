import * as React from 'react';
import {shallow} from 'enzyme';

import Unexist from './index';

describe('Unexist', () => {
  it('render', () => {
    const result = shallow(<Unexist/>);

    expect(result).toMatchSnapshot();
  });
});
