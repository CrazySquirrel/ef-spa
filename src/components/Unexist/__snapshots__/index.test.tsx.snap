version https://git-lfs.github.com/spec/v1
oid sha256:d7db8d66579463b1b88b51da735ed394b205537bf3e13165d605c52f644e5ab9
size 2056
